### DISCUSSION TITLE
Weird numbers when using a VPS based on xen technologies

### DISCUSSION THREAD

**Discussion by jdrouet** [hubblo-org/scaphandre#340] [General]
## Bug description

When I cat `/sys/devices/virtual/powercap/intel-rapl/intel-rapl:0/energy_uj` every seconds on my VPS I get the following result

```
194127997354
45766381128
45831036419
45895297790
194961270917
46025818025
195088717917
195154518652
195217628171
195283094556
195347616974
46422262201
46488823065
46554152305
195605807719
46686185292
195732606528
46819468154
195866360824
195930510501
195995971759
47080028827
47148009244
47213175459
47280270685
47345139416
47409635711
196448336263
196513124183
47604736713
196643367563
196707819058
```

We can see values that are no consistent.

## To Reproduce

- Take a vps based on xen technologies
- `cat` a `energy_uj` file every 1s seconds

## Expected behavior

Having a way to detect when the context switches and ignore the previous value.

## Environment

- Linux distribution version: Debian buster
- Kernel version: 4.19.0-14-amd64 (might explain why as well...)

**Comment by bpetit** [comment]
Hi @jdrouet,

This seems like a powercap-specific topic, but could be avery valuable insight on the Xen hypervisor behavior.

Could you share what provider you use ? 

I have honestly no idea why you seem to get values about two different sources. Maybe discussing about this with the provider could help ?

**Comment by jdrouet** [comment]
It's a VPS from ikoula

**Comment by bpetit** [comment]
Moving this to discussion.
